### PR TITLE
Support configuring retryOnConnectionFailure

### PR DESCRIPTION
**Pull request by tgregory-block**
Supports configuring `retryOnConnectionFailure`, an [OkHttpClient parameter](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/retry-on-connection-failure/) to automatically retry connecting to an external endpoint. The Wisp PR that [wires this up to OkHttp is here](https://github.com/cashapp/wisp/pull/88) My ultimate goal is to enable configuration via SkimConfig so we can pass this is via yaml. With that in mind, do I need to touch `HttpClientEndpointConfig` or `BackwardsCompatibleEndpointConfig`? 

I'm thinking not, since `protocols` was added without altering those, and that moving the endpoint's properties under `clientConfig` in our config files will bypass the backwards compat converter and let us use the new property?

### CODE REVIEW

**Review by frojasg — APPROVED** [review]
LGTM

File: misk-aws2-dynamodb-testing/build.gradle.kts [hunk 1]
```diff
@@ -25,6 +25,10 @@ dependencies {
   implementation("org.apache.logging.log4j:log4j-core:2.18.0")
   implementation("org.apache.logging.log4j:log4j-api:2.18.0")
 
+  // There is in incompatibility between com.amazonaws:DynamoDBLocal and recent ANTLR versions.
+  // We have to force down to a known-working version.
+  implementation("org.antlr:antlr4-runtime") { version { strictly("4.9.3") }}
```
Comment (tso) [inline misk-aws2-dynamodb-testing/build.gradle.kts:30]: Note that https://github.com/cashapp/misk/commit/12cb717a6e46fd6226feeabdc3f9fe8f555112ae added a check so you can remove these clamps

Comment (tgregory-block) [reply misk-aws2-dynamodb-testing/build.gradle.kts]: Thanks Tristan, reverted. And thanks for getting the buildkite stuff sorted 🙏

Comment (tso) [reply misk-aws2-dynamodb-testing/build.gradle.kts]: no problem! feel free to merge whenever

Comment (tgregory-block) [reply misk-aws2-dynamodb-testing/build.gradle.kts]: lol I can't, no write access. Are you able to merge/approve the test runs?

Comment (jdm-square) [reply misk-aws2-dynamodb-testing/build.gradle.kts]: I've pressed the buttons.

**Pull request merged**